python/imjoy_rpc/connection/socketio_connection.py

```diff
@@ -53,36 +50,34 @@ def register_codec(self, config):
 
         self._codecs[config["name"]] = dotdict(config)
 
-    def start(self, channel):
+    def start(self, url):
         sio = socketio.AsyncClient()
 
-        @sio.on("join_rpc_channel")
-        def join_rpc_channel(data):
-            self._create_new_connection(sio, data["sid"])
+        def registered(config):
+            @sio.on(config["channel"] + "-new-client")
+            def on_new_client(client_info):
+                self._create_new_connection(
+                    sio, config["channel"], client_info["channel"]
+                )
 
         @sio.event
         async def connect():
             await sio.emit("register_plugin", {"id": sio.sid}, callback=registered)
 
         self.sio = sio
-        print("plugin started.")
-        loop.run_until_complete(sio.wait())
-
-        if url:
-            logger.info("connecting to %s", url)
-            asyncio.ensure_future(sio.connect(url))
-
-        if wait:
-            loop = asyncio.get_event_loop()
-            loop.run_until_complete(sio.wait())
+        asyncio.ensure_future(self.sio.connect(url))
```

Comment:
Where is the event loop started? Is that the responsibility of the user of this library? In that case it might be more clear to make the interface `start` a coroutine function to require a running event loop.